source/common/stats/stats_impl.cc

```diff
@@ -150,10 +150,20 @@ bool TagExtractorImpl::extractTag(const std::string& stat_name, std::vector<Tag>
 }
 
 RawStatData* HeapRawStatDataAllocator::alloc(const std::string& name) {
-  // This must be zero-initialized
+  std::unique_lock<std::mutex> lock(mutex_);
+
+  absl::string_view key = name;
```

Comment:
nit: I don't think you need this line anymore since you can just pass name to initialize directly (string_view will implicitly be constructed from a string IIUC).